app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt

```diff
@@ -569,12 +569,13 @@ class StateFragmentPresenter @Inject constructor(
                 transaction
                   .add(dialogFragment, TAG_SURVEY_WELCOME_DIALOG)
                   .commitNow()
+
+                // Changes to underlying DataProviders will update the gating result.
+                // liveData.removeObserver(this)
```

Comment:
If this isn't needed, can the changes be reverted for this presenter?

Reply:
Done